src/types/dataModel/eventVolunteer.ts

```diff
@@ -23,13 +24,24 @@ const zEventVolunteerEntity = zEventVolunteer.extend({
   event: zObjectId,
 });
 
-export const zCreateEventVolunteerRequest = zEventVolunteer.extend({
+export const zCreateEventVolunteerRequestBase = zEventVolunteer.extend({
   organization: zObjectId.optional(),
-  volunteer: z.union([zObjectId, zCreateVolunteerRequest]),
   event: zObjectId,
+  volunteer: z.union([zObjectId, zCreateVolunteerRequest]),
   verifier: z.string().optional(),
 });
 
```

Comment:
```suggestion
 // If `isEdited` is false (when updating an existing volunteer during checkin), zCreateEventVolunteerRequest will just be zCreateEventVolunteerRequestBase.
 // Otherwise, the volunteer field will be an ObjectId, and `updatedVolunteer` will exist
```